apps/mobile/src/libs/components/checkbox-categories-form/checkbox-categories-form.tsx

```diff
@@ -0,0 +1,124 @@
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import { BaseColor, NumericalValue } from "~/libs/enums/enums";
+import {
+	useAppForm,
+	useCallback,
+	useEffect,
+	useFormController,
+} from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+import {
+	type CategoriesSelectedRequestDto,
+	type QuizScoresGetAllItemResponseDto,
+} from "~/libs/types/types";
+import { categoriesSelectedValidationSchema } from "~/packages/categories/categories";
+
+import {
+	CATEGORIES_FIELD_NAME,
+	CATEGORIES_FORM_DEFAULT_VALUES,
+	CHECK_ALL_CATEGORIES_NAME,
+} from "./libs/constants/constants";
+import {
+	getAllCategoryIds,
+	getCategoriesSortedByScore,
+	getSelectedCategoryIds,
+} from "./libs/helpers/helpers";
+
+type Properties = {
+	categories: QuizScoresGetAllItemResponseDto[];
+	onSubmit: (payload: CategoriesSelectedRequestDto) => void;
+	submitButtonName: string;
+};
+
+const CheckboxCategoriesForm: React.FC<Properties> = ({
+	categories,
+	onSubmit,
+	submitButtonName,
+}: Properties): JSX.Element | null => {
+	const { control, errors, handleSubmit, reset, setValue } = useAppForm<{
+		[CATEGORIES_FIELD_NAME]: number[];
+	}>({
+		defaultValues: {
+			[CATEGORIES_FIELD_NAME]: CATEGORIES_FORM_DEFAULT_VALUES,
+		},
+		validationSchema: categoriesSelectedValidationSchema,
+	});
+
+	useEffect(() => {
+		reset({
+			categoryIds: getSelectedCategoryIds(
+				getCategoriesSortedByScore(categories),
+			),
+		});
+	}, [reset, categories]);
+
+	const { field } = useFormController({ control, name: CATEGORIES_FIELD_NAME });
+	const { value } = field;
+
```

Comment:
Is it possible to use here `const { value, onChange } = field;` and replace the use of **setValue** with **onChange** everywhere?